math/src/field/extensions/quadratic.rs

```diff
@@ -313,6 +315,34 @@ impl<B: ExtensibleField<2>> From<u8> for QuadExtension<B> {
     }
 }
 
+impl<B: ExtensibleField<2>> TryFrom<u64> for QuadExtension<B> {
+    type Error = String;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let felt = B::try_from(value);
```

Comment:
minor nit: I'd probably name the variable `base_element` to more explicit. Or maybe just put this line directly into the match statement.

But also, can we not do `B::try_from(value)?;` to avoid the match statement below?

Reply:
I decided to use `B::try_from(value)?;` because without it I have to manage the field modulus `M` constant, which is different for different `B`, so I'm not sure that we can avoid it